source/opengl/opengl_effect_compiler.cpp

```diff
@@ -2226,7 +2226,7 @@ namespace reshade::opengl
 
 		auto &uniform_storage = _runtime->get_uniform_value_storage();
 
-		if (_uniform_storage_offset + _uniform_buffer_size >= uniform_storage.size())
+		if (_uniform_storage_offset + _uniform_buffer_size >= static_cast<ptrdiff_t>(uniform_storage.size()))
```

Comment:
The -1 initialization is a similar but different field in `runtime_objects.hpp`. I don't know why I made them `int`s, but I guess I was just testing this out and later kept the code without refactoring it properly.